File: src/store/message-identity.ts
import { createHash } from "node:crypto";

export function buildMessageIdentityKey(role: string, content: string): string {
  return `${role}\u0000${content}`;
}

export function buildMessageIdentityHash(role: string, content: string): string {
  return createHash("sha256")
    .update(role)
    .update("\u0000")
    .update(content)
    .digest("hex");
}

File: tui/message_identity.go
package main

import (
	"crypto/sha256"
	"encoding/hex"
)

func messageIdentityHash(role, content string) string {
	sum := sha256.New()
	sum.Write([]byte(role))
	sum.Write([]byte{0})
	sum.Write([]byte(content))
	return hex.EncodeToString(sum.Sum(nil))
}
